ops/testing.py

```diff
@@ -1080,6 +1097,16 @@ def get_pebble(self, socket_path: str):
         return client
 
     def planned_units(self):
+        """Simulate fetching the number of planned application units from the model.
+
+        If self._planned_units is None, then we simulate what the Juju controller will do, which is
+        to report the number of peers, plus one (we include this unit in the count). This can be
+        overridden for testing purposes: a charm author can set the number of planned units
+        explicitly by calling `Harness.set_planned_units`
+        """
+        if self._planned_units is not None:
+            return self._planned_units
```

Comment:
Should we have a way to set this back to None?
I guess Python doesn't care about args so `set_planned_units(None)` would do it. If we do want that to work, we should probably document it.

Reply:
I added a reset_planned_units method.